corehq/apps/app_manager/models.py

```diff
@@ -4776,18 +4776,18 @@ def set_form_versions(self):
         def _hash(val):
             return hashlib.md5(val).hexdigest()
 
-        previous_version = self.get_previous_version()
-        if previous_version:
-            force_new_version = self.build_profiles != previous_version.build_profiles
+        latest_build = self.get_latest_build()
+        if latest_build:
```

Comment:
nit: you could short circuit this logic to remove a level of indentation
```py
if not latest_build:
    return None
...
```

Reply:
https://github.com/dimagi/commcare-hq/pull/24860/commits/35967f09cce8dced47dc5d596b6212f205552dc7